features/step_definitions/purap/requisition.rb

```diff
@@ -0,0 +1,152 @@
+    item_tab.account_number_new_search(new_item_index)
+    on AccountLookupPage do |acct_lookup|
+      acct_lookup.sub_fund_group_code.set (get_aft_parameter_values_as_hash(ParameterConstants::DEFAULTS_FOR_ITEMS))[:account_subfund_lookup]
+      acct_lookup.search
+      acct_lookup.wait_for_search_results
+      acct_lookup.return_random
+    end
+    # Attempt a set number of times to obtain a random object code with the AFT parameter specified Object Level
+    # value that does not cause a business rule error for the Object Sub-Type code assigned to the random Object Code
+    # using the default Chart and Percent values.
+    max_num_searches = 10
+    number_object_search_attempts = 0
+    begin
+      item_tab.object_code_new_search(new_item_index)
+      on ObjectCodeLookupPage do |obj_lookup|
+        obj_lookup.level_code.set (get_aft_parameter_values_as_hash(ParameterConstants::DEFAULTS_FOR_ITEMS))[:object_level_lookup]
+        obj_lookup.search
+        obj_lookup.wait_for_search_results
+        obj_lookup.return_random
+      end
+      number_object_search_attempts +=1
+      # business rules are verified when add button is pressed
+      item_tab.add_item_accounting_line(new_item_index)
+      page_errors = $current_page.errors
+    end while not(page_errors == []) and number_object_search_attempts < max_num_searches #while-object code business rule failure
+    # Get page data in backing object prior to failure check so it is known why should failure occur
+    @requisition.items.update_from_page! :new
+    fail ArgumentError, "Attempted #{max_num_searches} times to find Object Code with an Object Level of #{((get_aft_parameter_values_as_hash(ParameterConstants::DEFAULTS_FOR_ITEMS))[:object_level_lookup])} but each Object Code selected generated a business rule error when attempting to add the Account." if number_object_search_attempts == max_num_searches
+  end
+end
+
+
+Then /^a Commodity Reviewer does (.*) a Pending or Future Action approval request for the (sensitive|non\-sensitive) Commodity Code$/ do |existence_check, commodity_code_type|
+  # verify that Pending Action Requests OR Future Action Requests (do|do not) show routing to Commodity Reviewer
+  role_exists_in_pending = nil
+  role_exists_in_future = nil
+  pending_cr_row = -1
+  future_cr_row =  -1
+
+  on(KFSBasePage) do |page|
+    page.expand_all
+    page.show_pending_action_requests if page.show_pending_action_requests_button.exists?
+    page.show_future_action_requests if page.show_future_action_requests_button.exists?
+
+    annotation_pending_col = page.pnd_act_req_table.keyed_column_index(:annotation)
+    pending_cr_row = page.pnd_act_req_table
+                         .column(annotation_pending_col)
+                         .index{ |c| c.exists? && c.visible? && c.text.match(/Commodity Reviewer/) }
+    !(pending_cr_row == -1 or pending_cr_row.nil?) ? role_exists_in_pending = true : role_exists_in_pending = false
```

Comment:
This line would be easier to read as a regular if-statement instead of a ternary. Further, this is equivalent, correct?
```ruby
role_exists_in_pending = !(pending_cr_row == -1 || pending_cr_row.nil?)
```

Also, `or` is not equivalent to `||`.